### PR TITLE
formiko: init at 1.4.3

### LINKED ISSUE #112998: Package request: Formiko
**Project description**

> Formiko is reStructuredText and MarkDown editor and live previewer. It is written in Python with Gtk3, GtkSourceView and Webkit2.

The application is available [in Ubuntu repository](https://packages.ubuntu.com/en/source/focal/formiko) and [on Flathub](https://flathub.org/apps/details/cz.zeropage.Formiko).

**Metadata**

* homepage URL: https://github.com/ondratu/formiko
* source URL: https://github.com/ondratu/formiko
* license: bsd
* platforms: unix, linux, darwin

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Closes https://github.com/NixOS/nixpkgs/issues/112998

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113141` at 87f66aba run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/30a56865b1868b71b5a3f0298668c5c6">formiko</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `pkg-config`.
Near pkgs/applications/editors/formiko/default.nix:8:3:
```
  |
8 | , pkg-config
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `python3`.
Near pkgs/applications/editors/formiko/default.nix:5:3:
```
  |
5 | , python3
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/applications/editors/formiko/default.nix:1:3:
```
  |
1 | { stdenv
  |   ^
```
</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 113141` at 87f66aba run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/93420fa78ff90212173030e43d5fe438">formiko</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `pkg-config`.
Near pkgs/applications/editors/formiko/default.nix:8:3:
```
  |
8 | , pkg-config
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `python3`.
Near pkgs/applications/editors/formiko/default.nix:5:3:
```
  |
5 | , python3
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/applications/editors/formiko/default.nix:1:3:
```
  |
1 | { stdenv
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/editors/formiko/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    gtk3
+    gtksourceview
+    gtkspell3
+    librsvg
+    pygobject3
+    webkitgtk
+  ];
+
+  # Needs a display
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/formiko/default.nix:44]: Can we use a fake display with xvfb run?

File: pkgs/applications/editors/formiko/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+, gtksourceview
+, gtkspell3
+, librsvg
+, pygobject3
+, webkitgtk
+}:
+
+buildPythonApplication rec {
+
+  pname = "formiko";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/formiko/default.nix:17]:
```suggestion
buildPythonApplication rec {
  pname = "formiko";
```

File: pkgs/applications/editors/formiko/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  meta = with lib; {
+    description = "reStructuredText editor and live previewer";
+    homepage = "https://github.com/ondratu/formiko";
+    license = licenses.bsd2; # May be wrong
+    maintainers = with maintainers; [ shamilton ];
+    platforms = platforms.linux;
+  };
+}
+
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/formiko/default.nix:54]: Final new line to much.

File: pkgs/applications/editors/formiko/default.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    webkitgtk
+  ];
+
+  # Needs a display
+  doCheck = false;
+
+  meta = with lib; {
+    description = "reStructuredText editor and live previewer";
+    homepage = "https://github.com/ondratu/formiko";
+    license = licenses.bsd2; # May be wrong
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/formiko/default.nix:49]:
```suggestion
    license = licenses.bsd3;
```

https://metadata.ftp-master.debian.org/changelogs//main/f/formiko/formiko_1.3.0-1_copyright

File: pkgs/applications/editors/formiko/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/editors/formiko/default.nix:44]: It works but it seems like there isn't any test suite, it's just trying to launch the app and the build just waits for someone to close it (which will never happen).

File: pkgs/applications/editors/formiko/default.nix (hunk 3 above)
Comment (SCOTT-HAMILTON) [reply pkgs/applications/editors/formiko/default.nix]: Fixed

**Pull request merged**